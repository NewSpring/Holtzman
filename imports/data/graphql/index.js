
import { Meteor } from "meteor/meteor";
// eslint-disable-next-line
import { DDP } from "meteor/ddp";
import { Accounts } from "meteor/accounts-base";

import ApolloClient, { createNetworkInterface } from "apollo-client";

const networkInterface = createNetworkInterface({ uri: Meteor.settings.public.heighliner });

const authMiddleware = {
  applyMiddleware(req, next) {
    const request = req;
    let currentUserToken;

    // on the server
    // eslint-disable-next-line
    const status = DDP._CurrentInvocation.get();
    if (status && status.connection && status.connection.id) {
      // eslint-disable-next-line
      currentUserToken = Accounts._getLoginToken(status.connection.id);
    }

    // on the client
    // eslint-disable-next-line
    if (Accounts._storedLoginToken) currentUserToken = Accounts._storedLoginToken();
    if (!currentUserToken) {
      next();
      return;
    }

    if (!request.options.headers) {
      if (fetch.Headers) {
        request.options.headers = new fetch.Headers();
      } else {
        request.options.headers = new Headers();
      }
    }

    request.options.headers.Authorization = currentUserToken;

    next();
  },
};

networkInterface.use([
  authMiddleware,
]);

const GraphQL = new ApolloClient({
  networkInterface,
  ssrMode: Meteor.isServer,
  shouldBatch: false, // XXX not working with multiple root fields on a query
});

export default GraphQL;

export {
  GraphQL,
  networkInterface,
  authMiddleware,
};
