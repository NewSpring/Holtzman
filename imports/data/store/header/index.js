/*
  Share action types
*/
import reducer from "./reducer";
import "./saga";
import { addReducer } from "../utilities";

addReducer({
  header: reducer,
});

export default {
  show: (options = {}) => ({
    type: "HEADER.TOGGLE_VISIBILITY",
    visible: typeof options.visible === "undefined" ? true : options.visible,
    statusBar: options.statusBar,
  }),
  hide: (options = {}) => ({
    type: "HEADER.TOGGLE_VISIBILITY",
    visible: typeof options.visible === "undefined" ? false : options.visible,
    statusBar: options.statusBar,
  }),

  set: (content, requestee) => ({ type: "HEADER.SET", content, requestee }),
  color: (color) => ({ type: "HEADER.SET", content: { color } }),
  title: (title) => ({ type: "HEADER.SET", content: { title } }),
  light: (light) => ({ type: "HEADER.SET", content: { light } }),
  isSearch: (isSearch, searchSubmit) => ({
    type: "HEADER.SET",
    content: { isSearch, searchSubmit },
  }),

  lock: (requestee) => ({ type: "HEADER.LOCK", requestee }),
  unlock: () => ({ type: "HEADER.UNLOCK" }),

  // XXX this currently doesn't affect the store
  statusBarColor: (color) => ({ type: "STATUSBAR.SET", color }),
};
