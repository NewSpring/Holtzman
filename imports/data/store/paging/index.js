import reducer from "./reducer";
import { addReducer } from "../utilities";

addReducer({
  paging: reducer,
});

export default {
  increment: () => ({ type: "PAGING.INCREMENT" }),
  pause: () => ({ type: "PAGING.PAUSE" }),
  resume: () => ({ type: "PAGING.RESUME" }),
  reset: () => ({ type: "PAGING.RESET" }),
};
