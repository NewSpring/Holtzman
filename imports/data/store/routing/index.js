
// import "./saga"

// Constants
// taken from https://github.com/reactjs/react-router-redux/blob/master/src/index.js
const TRANSITION = "@@router/TRANSITION";
const UPDATE_LOCATION = "@@router/UPDATE_LOCATION";

const SELECT_LOCATION = (state) => state.routing.location;

function transition(method) {
  return (...args) => ({
    type: TRANSITION,
    payload: { method, args },
  });
}

export const push = transition("push");
export const replace = transition("replace");
export const go = transition("go");
export const goBack = transition("goBack");
export const goForward = transition("goForward");

export const routeActions = { push, replace, go, goBack, goForward };

function updateLocation(location) {
  return {
    type: UPDATE_LOCATION,
    payload: location,
  };
}

// Reducer

const initialState = {
  location: undefined,
};

export function routeReducer(state = initialState, { type, payload: location }) {
  if (type !== UPDATE_LOCATION) {
    return state;
  }

  return { ...state, location };
}

// Syncing

export function syncHistory(history) {
  let unsubscribeHistory;
  let currentKey;
  let unsubscribeStore;

  let connected = false;
  let syncing = false;

  history.listen((location) => { initialState.location = location; })();

  function middleware(store) {
    unsubscribeHistory = history.listen((location) => {
      currentKey = location.key;
      if (syncing) {
        // Don't dispatch a new action if we're replaying location.
        return;
      }
      const { routing } = store.getState();
      if (!location.previous) {
        location.previous = []; // eslint-disable-line no-param-reassign
      }
      if (!routing.location.previous) {
        routing.location.previous = [];
      }

      if (routing.location.previous[routing.location.previous.length - 1] === location.pathname) {
        routing.location.previous.splice(-1);
        location.previous = routing.location.previous; // eslint-disable-line no-param-reassign
      } else {
        // eslint-disable-next-line no-param-reassign
        location.previous = [...routing.location.previous, ...[routing.location.pathname]];
      }


      store.dispatch(updateLocation(location));
    });

    connected = true;

    // eslint-disable-next-line consistent-return
    return (next) => (action) => {
      if (action.type !== TRANSITION || !connected) {
        return next(action);
      }

      const { payload: { method, args } } = action;

      history[method](...args);
    };
  }

  middleware.listenForReplays =
    (store, selectLocationState = SELECT_LOCATION) => {
      const getLocationState = () => selectLocationState(store.getState());
      const initialLocation = getLocationState();

      unsubscribeStore = store.subscribe(() => {
        const location = getLocationState();

        // If we're resetting to the beginning, use the saved initial value. We
        // need to dispatch a new action at this point to populate the store
        // appropriately.
        if (location.key === initialLocation.key) {
          history.replace(initialLocation);
          return;
        }

        // Otherwise, if we need to update the history location, do so without
        // dispatching a new action, as we're just bringing history in sync
        // with the store.
        if (location.key !== currentKey) {
          syncing = true;
          history.transitionTo(location);
          syncing = false;
        }
      });
    };

  middleware.unsubscribe = () => {
    unsubscribeHistory();
    if (unsubscribeStore) {
      unsubscribeStore();
    }

    connected = false;
  };

  return middleware;
}
