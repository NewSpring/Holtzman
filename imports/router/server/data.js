// stolen from https://github.com/kadirahq/flow-router/blob/ssr/server/ssr_data.js

export default function patchSubscribeData(ReactRouterSSR) {
  const originalSubscribe = Meteor.subscribe;

  Meteor.subscribe = function (pubName) {
    const params = Array.prototype.slice.call(arguments, 1);

    const ssrContext = ReactRouterSSR.ssrContext.get();
    if (ssrContext) {
      ReactRouterSSR.inSubscription.withValue(true, () => {
        ssrContext.addSubscription(pubName, params);
      });
    }

    if (originalSubscribe) {
      originalSubscribe.apply(this, arguments);
    }

    return {
      ready: () => true,
    };
  };

  const Mongo = Package.mongo.Mongo;
  const originalFind = Mongo.Collection.prototype.find;

  Mongo.Collection.prototype.find = function (selector = {}, options = {}) {
    selector = selector || {};
    const ssrContext = ReactRouterSSR.ssrContext.get();
    if (ssrContext && !ReactRouterSSR.inSubscription.get()) {
      const collName = this._name;

      // this line is added just to make sure this works CollectionFS
      if (typeof this._transform === "function") {
        options.transform = this._transform;
      }

      const collection = ssrContext.getCollection(collName);
      const cursor = collection.find(selector, options);
      return cursor;
    }

    return originalFind.call(this, selector, options);
  };

  // We must implement this. Otherwise, it'll call the origin prototype's
  // find method
  Mongo.Collection.prototype.findOne = function (selector, options) {
    options = options || {};
    options.limit = 1;
    return this.find(selector, options).fetch()[0];
  };

  const originalAutorun = Tracker.autorun;

  Tracker.autorun = (fn) => {
    // if autorun is in the ssrContext, we need fake and run the callback
    // in the same eventloop
    if (ReactRouterSSR.ssrContext.get()) {
      const c = { firstRun: true, stop: () => {} };
      fn(c);
      return c;
    }

    return originalAutorun.call(Tracker, fn);
  };

  // By default, Meteor[call,apply] also inherit SsrContext
  // So, they can't access the full MongoDB dataset because of that
  // Then, we need to remove the SsrContext within Method calls
  ["call", "apply"].forEach((methodName) => {
    const original = Meteor[methodName];
    Meteor[methodName] = (...args) => {
      const response = ReactRouterSSR.ssrContext.withValue(null, () => {
        return original.apply(this, args);
      });

      return response;
    };
  });

  // This is not available in the server. But to make it work with SSR
  // We need to have it.
  Meteor.loggingIn = () => {
    return false;
  };
}
