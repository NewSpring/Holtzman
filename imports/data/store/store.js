import "regenerator-runtime/runtime";

import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import { ApolloProvider } from "react-apollo";
import createSagaMiddleware from "redux-saga";
import reduxReset from "redux-reset";

import GraphQL from "../graphql";

import { reducers, middlewares, sagas } from "./utilities";
import { syncHistory, routeReducer } from "./routing";

const createReduxStore = (initialState, history) => {
  if (initialState) {
    // bug with SSR
    delete initialState.nav; // eslint-disable-line no-param-reassign
  }

  const joinedReducers = { ...reducers,
    ...{
      routing: routeReducer,
      apollo: GraphQL.reducer(),
    },
  };

  const sharedMiddlewares = [...middlewares, GraphQL.middleware()];

  const reduxRouterMiddleware = syncHistory(history);

  const sagaMiddleware = createSagaMiddleware();

  let sharedCompose = [
    applyMiddleware(
      ...sharedMiddlewares,
      sagaMiddleware,
      reduxRouterMiddleware,
    ),
    reduxReset(),
  ];

  if (process.env.NODE_ENV !== "production") {
    sharedCompose = [...sharedCompose, ...[
      typeof window === "object" && typeof window.devToolsExtension !== "undefined" ?
        window.devToolsExtension() :
        f => f,
    ]];
  }

  const store = compose(...sharedCompose)(createStore)(
    combineReducers(joinedReducers), initialState,
  );

  sagas.forEach(saga => sagaMiddleware.run(saga()));

  return store;
};


const wrapper = ApolloProvider;

export {
  wrapper,
  createReduxStore,
};
