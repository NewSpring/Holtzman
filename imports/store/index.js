
import accounts from "./accounts";
import audio from "./audio";
import give from "./give";
import liked from "./liked";
import live from "./live";
import modal from "./modal";
import nav from "./nav";
import paging from "./paging";
import responsive from "./responsive";
import routing from "./routing";
import search from "./search";
import sections from "./sections";
import share from "./share";
import topics from "./topics";
// import fullscreen from "./fullscreen";

let header;
if (process.env.NATIVE) header = require("./header").default;

import { wrapper, createReduxStore } from "./store";
import { addMiddleware, addReducer, createReducer, addSaga } from "./utilities";

export {
  accounts,
  audio,
  give,
  header,
  liked,
  live,
  modal,
  nav,
  paging,
  responsive,
  routing,
  search,
  sections,
  share,
  topics,

  wrapper,
  createReduxStore,

  addMiddleware,

  addReducer,
  createReducer,
};
