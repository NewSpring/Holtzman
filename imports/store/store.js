import "regenerator-runtime/runtime";

import { Component, PropTypes } from "react";
import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import { ApolloProvider } from "react-apollo";
import createSagaMiddleware from "redux-saga";
import reduxReset from "redux-reset";

import { GraphQL } from "../graphql";

import { reducers, middlewares, sagas } from "./utilities";
import { syncHistory, routeReducer } from "./routing";

const createReduxStore = (initialState, history) => {

  if (initialState) {
    // bug with SSR
    delete initialState.nav;
  }

  const joinedReducers = {...reducers, ...{
    routing: routeReducer,
    apollo: GraphQL.reducer(),
  }};

  let sharedMiddlewares = [...middlewares, ...GraphQL.middleware()];

  const reduxRouterMiddleware = syncHistory(history);

  const sagaMiddleware = createSagaMiddleware();
  let sharedCompose = [
    applyMiddleware(
      ...sharedMiddlewares,
      sagaMiddleware,
      reduxRouterMiddleware
    ),
    reduxReset(),
  ];

  if (process.env.NODE_ENV != "production") {
    sharedCompose = [...sharedCompose, ...[
      typeof window === "object" && typeof window.devToolsExtension !== "undefined" ? window.devToolsExtension() : f => f
    ]];
  }

  const store = compose(...sharedCompose)(createStore)(
    combineReducers(joinedReducers), initialState
  );

  sagas.forEach(saga => sagaMiddleware.run(saga()));

  return store;

};


const wrapper = ApolloProvider;

export {
  wrapper,
  createReduxStore
};
