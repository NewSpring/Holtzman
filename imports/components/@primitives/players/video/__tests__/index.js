import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { audio as audioActions } from "../../../../../data/store";
import { VideoPlayerWithoutData as VideoPlayer } from "../";

window.Wistia = {
  ready: jest.fn(),
  Player: {
    create: jest.fn(),
  },
};

const defaultProps = {
  id: "2",
  hide: false,
  success: jest.fn(),
  style: {
    color: "red",
  },
  audioState: "default",
  dispatch: jest.fn(),
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <VideoPlayer {...newProps} />;
};

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

xit("calls createPlayer on mount", () => {
  const wrapper = shallow(generateComponent());
  const mockCreatePlayer = jest.fn();
  wrapper.instance().createPlayer = mockCreatePlayer;
  wrapper.instance().componentDidMount();
  expect(mockCreatePlayer).toHaveBeenCalledTimes(1);
  expect(mockCreatePlayer).toHaveBeenCalledWith(defaultProps.id, defaultProps.success);
});

xit("calls setEmbedCode if current id is not next id", () => {
  const wrapper = shallow(generateComponent());
  const mockSetEmbedCode = jest.fn();
  wrapper.instance().player = {
    setEmbedCode: mockSetEmbedCode,
  };
  wrapper.setProps({ id: "3" });
  expect(mockSetEmbedCode).toHaveBeenCalledTimes(1);
  expect(mockSetEmbedCode).toHaveBeenCalledWith("3");
});

xit("calls pause if next state is playing and current state is default", () => {
  const wrapper = shallow(generateComponent());
  const mockPause = jest.fn();
  wrapper.instance().player = {
    pause: mockPause,
  };
  wrapper.setProps({ audioState: "playing" });
  expect(mockPause).toHaveBeenCalledTimes(1);
});

xit("calls pause if next state is loading and current state is default", () => {
  const wrapper = shallow(generateComponent());
  const mockPause = jest.fn();
  wrapper.instance().player = {
    pause: mockPause,
  };
  wrapper.setProps({ audioState: "loading" });
  expect(mockPause).toHaveBeenCalledTimes(1);
});

xit("Removes the player on unmount", () => {
  const wrapper = shallow(generateComponent());
  const mockRemove = jest.fn();
  wrapper.instance().player = {
    remove: mockRemove,
  };
  wrapper.unmount();
  expect(mockRemove).toHaveBeenCalledTimes(1);
});

xit("createPlayer sets timeout to try again if no OO", () => {
  jest.useFakeTimers();
  window.OO = null;
  const mockCallback = jest.fn();
  const wrapper = shallow(generateComponent());
  wrapper.instance().createPlayer("2", mockCallback);
  const mockCreatePlayer = jest.fn();
  wrapper.instance().createPlayer = mockCreatePlayer;
  jest.runAllTimers();
  expect(mockCreatePlayer).toHaveBeenCalled();
});

// XXX yep
xit("createPlayer sets up player", () => {
  jest.useFakeTimers();
  const mockReady = jest.fn();
  const mockCreate = jest.fn(() => {});
  window.OO = {
    ready: mockReady,
    Player: {
      create: mockCreate,
    },
    EVENTS: {
      PLAYED: "PLAYED",
      PLAY: "PLAY",
      PLAY_FAILED: "PLAY_FAILED",
      FULLSCREEN_CHANGED: "FULLSCREEN_CHANGED",
    },
  };
  const mockDispatch = jest.fn();
  audioActions.pause = jest.fn();
  const wrapper = shallow(generateComponent({
    dispatch: mockDispatch,
  }));
  const mockCallback = jest.fn();
  wrapper.instance().createPlayer("2", mockCallback);
  expect(mockReady).toHaveBeenCalled();
  // simulate OO.ready
  mockReady.mock.calls[0][0]();
  expect(mockCreate).toHaveBeenCalled();
  expect(mockCreate.mock.calls[0][0]).toBe("ooyala-player-2");
  expect(mockCreate.mock.calls[0][1]).toBe("2");
  // simulate onCreate callback
  const mockPlayer = {
    isPlaying: jest.fn(() => true),
    isFullscreen: jest.fn(() => false),
    mb: {
      subscribe: jest.fn(),
    },
  };
  mockCreate.mock.calls[0][2].onCreate(mockPlayer);
  expect(mockDispatch).toHaveBeenCalledTimes(1);
  expect(audioActions.pause).toHaveBeenCalledTimes(1);
  // expect(mockCallback).toHaveBeenCalledTimes(1);
  expect(mockPlayer.mb.subscribe).toHaveBeenCalledTimes(4);
  const mockDestroy = jest.fn();
  wrapper.instance().destroy = mockDestroy;
  // PLAYED callback
  expect(mockPlayer.mb.subscribe.mock.calls[0][0]).toBe(window.OO.EVENTS.PLAYED);
  expect(mockPlayer.mb.subscribe.mock.calls[0][1]).toBe("Video");
  mockPlayer.mb.subscribe.mock.calls[0][2]();
  expect(mockDestroy).toHaveBeenCalled();
  // PLAY callback
  expect(mockPlayer.mb.subscribe.mock.calls[1][0]).toBe(window.OO.EVENTS.PLAY);
  expect(mockPlayer.mb.subscribe.mock.calls[1][1]).toBe("Video");
  mockPlayer.mb.subscribe.mock.calls[1][2]();
  expect(mockDispatch).toHaveBeenCalledTimes(2);
  expect(audioActions.pause).toHaveBeenCalled();
  // PLAY_FAILED callback
  expect(mockPlayer.mb.subscribe.mock.calls[2][0]).toBe(window.OO.EVENTS.PLAY_FAILED);
  expect(mockPlayer.mb.subscribe.mock.calls[2][1]).toBe("Video");
  mockPlayer.mb.subscribe.mock.calls[2][2]();
  expect(mockDestroy).toHaveBeenCalled();
  // FULLSCREEN_CHANGED callback
  global.StatusBar = {
    styleLightContent: jest.fn(),
  };
  mockPlayer.mb.subscribe.mock.calls[3][2]();
  jest.runAllTimers();
  expect(global.StatusBar.styleLightContent).toHaveBeenCalled();
});

xit("show calls createPlayer if no player", () => {
  const wrapper = shallow(generateComponent());
  const mockCreatePlayer = jest.fn();
  wrapper.instance().createPlayer = mockCreatePlayer;
  wrapper.instance().show();
  expect(mockCreatePlayer).toHaveBeenCalledTimes(1);
  expect(mockCreatePlayer.mock.calls[0][0]).toBe(defaultProps.id);
  // simulate player ready callback
  wrapper.setState({ hide: true });
  mockCreatePlayer.mock.calls[0][1]();
  expect(wrapper.state().hide).toBe(false);
});

xit("show calls createPlayer if destroyed player", () => {
  const wrapper = shallow(generateComponent());
  const mockCreatePlayer = jest.fn();
  wrapper.instance().createPlayer = mockCreatePlayer;
  wrapper.instance().player = {
    state: "destroyed",
  };
  wrapper.instance().show();
  expect(mockCreatePlayer).toHaveBeenCalled();
  expect(mockCreatePlayer.mock.calls[0][0]).toBe(defaultProps.id);
  // simulate player ready callback
  wrapper.setState({ hide: true });
  mockCreatePlayer.mock.calls[0][1]();
  expect(wrapper.state().hide).toBe(false);
});

xit("show pauses audio", () => {
  const mockDispatch = jest.fn();
  audioActions.pause = jest.fn();
  const wrapper = shallow(generateComponent({
    dispatch: mockDispatch,
  }));
  wrapper.setState({ hide: true });
  // const mockPlay = jest.fn();
  // const mockPlayer = {
  //   play: mockPlay,
  // };
  // wrapper.instance().player = mockPlayer;
  wrapper.instance().show();
  expect(mockDispatch).toHaveBeenCalledTimes(1);
  expect(audioActions.pause).toHaveBeenCalledTimes(1);
  // expect(mockPlay).toHaveBeenCalledTimes(1);
  expect(wrapper.state().hide).toBe(false);
});

xit("hide updates state", () => {
  const wrapper = shallow(generateComponent());
  // const mockPause = jest.fn();
  // const mockPlayer = {
  //   pause: mockPause,
  // };
  // wrapper.instance().player = mockPlayer;
  wrapper.instance().hide();
  // expect(mockPause).toHaveBeenCalledTimes(1);
  expect(wrapper.state().hide).toBe(true);
});

xit("styles has non hidden styles", () => {
  const wrapper = shallow(generateComponent());
  expect(wrapper.instance().styles()).toMatchSnapshot();
});

xit("styles has hidden styles", () => {
  const wrapper = shallow(generateComponent());
  wrapper.setState({ hide: true });
  expect(wrapper.instance().styles()).toMatchSnapshot();
});
