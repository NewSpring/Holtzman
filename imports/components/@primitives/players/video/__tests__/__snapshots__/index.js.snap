// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with props 1`] = `
<div
  className="wistia_responsive_padding"
  style={
    Object {
      "padding": "56.25% 0 28px 0",
      "position": "relative",
    }
  }
>
  <div
    className="wistia_responsive_wrapper"
    style={
      Object {
        "height": "100%",
        "left": "0",
        "position": "absolute",
        "top": "0",
        "width": "100%",
      }
    }
  >
    <div
      className="wistia_embed wistia_async_2 videoFoam=true"
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      Â 
    </div>
  </div>
</div>
`;

exports[`styles has hidden styles 1`] = `
Object {
  "color": "red",
  "display": "none",
}
`;

exports[`styles has non hidden styles 1`] = `
Object {
  "color": "red",
}
`;
